AlphaWallet/ScrollableSegmentedControl/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,270 @@
+//
+//  ScrollableSegmentedControl.swift
+//  AlphaWallet
+//
+//  Created by Jerome Chan on 27/12/21.
+//
+
+import UIKit
+
+struct ScrollableSegmentedControlConfiguration {
+    let lineConfiguration: ScrollableSegmentedControlHighlightableLineViewConfiguration
+    let isProportionalWidth: Bool
+    let cellSpacing: CGFloat
+    let alignmentWhenNotScrollable: ScrollableSegmentedControl.Alignment
+    let animationDuration: TimeInterval
+    let animationCurve: UIView.AnimationCurve
+}
+
+class ScrollableSegmentedControl: UIControl {
+    enum Selection {
+        case unselected
+        case selected(Int)
+    }
+    // Only if not scrollable (total cells width + spacing < control width).
+    enum Alignment {
+        case leading
+        case trailing
+        case centered
+        case filled
+    }
+    // MARK: - Properties
```

Comment:
I don't think `MARK`s works in practice.. eg. a few lines later, what if a propery is both "UI Elements" and `Public`? (I know it looks better in Xcode)

And what if someone places a function under the wrong group (based on `MARK`)?

You can leave them in if they really help you though.

Reply:
It is used for quick navigation in Xcode.